File: project/angular_project/src/app/models/address.ts
import { City } from "./city";



export class Address {
  address_id: number;
  street: string;
  number: number;
  is_HQ: boolean;
  city: City;

  constructor() {
    this.address_id = 0;
    this.street = '';
    this.number = 0;
    this.is_HQ = false;
    this.city = new City();
  }
}

File: src/main/java/com/example/project/entities/Address.java
package com.example.project.entities;

import java.io.Serializable;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document(collection = "Address")
public class Address implements Serializable{
	@Id
	private int address_id;
	private String street;
	private int number;
	private Boolean is_HQ;
    private City city;
	
}